software/base/src/main/java/org/apache/brooklyn/entity/software/base/VanillaWindowsProcessWinRmDriver.java

```diff
@@ -49,15 +50,29 @@ public void start() {
 
     @Override
     public void install() {
+        // TODO: At some point in the future, this should probably be refactored to get the name of the machine in WinRmMachineLocation and set it as the hostname sensor
+        String hostname = null;
+        if (entity.getConfig(VanillaWindowsProcess.INSTALL_REBOOT_REQUIRED)) {
+            WinRmExecuteHelper checkHostnameTask = newScript("Checking hostname")
+                    .setCommand("hostname")
+                    .failOnNonZeroResultCode()
+                    .gatherOutput();
+            if (checkHostnameTask.execute() != 0) {
+                LOG.warn("Cannot determine hostname for entity {} machine {}", getEntity(), getLocation());
+            }
+            hostname = Strings.trimEnd(checkHostnameTask.getResultStdout());
```

Comment:
@aledsage I find it as the most reasonable way to get the actual hostname.
The `LOG.warn` statement which was there is not valid.
`.failOnNonZeroResultCode()` predicate will be used from the WinRmExecuteHelper to throw an exception if it is not valid.
I expect that this command should always succeed when winrm-ing to a working windows machine.